core/exception_vectors.S

```diff
@@ -491,7 +530,15 @@ NMIExceptionHandler:
         .balign  4
 UserExceptionHandler:
         .type   UserExceptionHandler, @function
-        xsr     a0, excsave1    # a0 now contains sp
+        // save a0, a1 to debug_saved_ctx before stack pointer is affected
```

Comment:
Debugger is triggered on UART interrupt, so when user calls breaks the execution, the interrupt is processed through LUserHandleInterrupt (which then calls C functions — thus the need to save a0 and a1). The rest of registers seems to be saved by xthal. Please correct me if I'm wrong, but it looks like when LUserHandleInterrupt is called, original values of a0 and a1 are lost, so these two must be saved right on the user exception entry,